webapp/src/components/sidebar_buttons/sidebar_buttons.jsx

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Tooltip, OverlayTrigger} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {RHSStates} from '../../constants';
+
+export default class SidebarButtons extends React.PureComponent {
+    static propTypes = {
+        theme: PropTypes.object.isRequired,
+        isTeamSidebar: PropTypes.bool,
+        showRHSPlugin: PropTypes.func.isRequired,
+        issues: PropTypes.arrayOf(PropTypes.object),
+        inIssues: PropTypes.arrayOf(PropTypes.object),
+        outIssues: PropTypes.arrayOf(PropTypes.object),
+        actions: PropTypes.shape({
+            list: PropTypes.func.isRequired,
+            updateRhsState: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            refreshing: false,
+        };
+    }
+
+    componentDidUpdate() {
+        this.getData();
+    }
```

Comment:
Why this? My first thought regarding this is that when we get the data, as we finish, we change the refreshing back to false. This will create a new render, which will call this componentDidUpdate, which will call get data, which will change back the state to true, making a new rendering.
Maybe I am missing something, but does look like a dangerous loop.

Reply:
You're right! I will add check with `prevProps` and `this.props` to resolve this.